src/main/java/seedu/address/logic/commands/EditByNameCommand.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.Arrays;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+import seedu.address.model.person.NameContainsAllKeywordsPredicate;
+import seedu.address.model.person.Person;
+
+//@@author zioul123
+/**
+ * Edits the details of an existing person in the address book.
+ */
+public class EditByNameCommand extends EditCommand {
+
+    protected final String personIdentifier;
+
+    public EditByNameCommand(String personIdentifier, EditPersonDescriptor editPersonDescriptor) {
+        super(Index.fromZeroBased(NO_INDEX), editPersonDescriptor);
+        requireNonNull(personIdentifier);
+
+        this.personIdentifier = personIdentifier;
+    }
+
+    //@@author zioul123-reused
+    //Adapted from the execute method of EditCommand.
```

Comment:
Don't have to give credit to using AB4 code (nus-cs2103-AY1819S1/forum#90 (comment))